packages/wallet-adapter/base/src/adapter.ts

```diff
@@ -193,8 +198,8 @@ export class StandardWalletAdapter extends BaseWalletAdapter implements Standard
     }
 
     #changed: StandardEventsListeners['change'] = (properties) => {
-        // If the adapter isn't connected or the change doesn't include accounts, do nothing.
-        if (!this.#account || !this.#publicKey || !('accounts' in properties)) return;
+        // If the adapter is disconnecting, or isn't connected, or the change doesn't include accounts, do nothing.
+        if (this.#disconnecting || !this.#account || !this.#publicKey || !('accounts' in properties)) return;
```

Comment:
Are we worried about losing the call to `this.#disconnected()` and `this.emit('disconnect')`? 

Or is `await this.#wallet.features[StandardDisconnect].disconnect();` in `disconnect` expected to include these?

Reply:
We still get those when we call `adapter.disconnect()` on L165-166:
```ts
        this.#disconnected();
        this.emit('disconnect');
```

This actually fixes another subtler issue which is that, when disconnecting, you'd be likely to get two `disconnect` events (one from `disconnect`, the other from `#changed`).